SwiftDraft/ReliantFramework/ReliantFrameworkTests/ReliantFrameworkTests.swift

```diff
@@ -54,17 +54,25 @@ struct ContextNeedingContext : ReliantContext {
         return ContextNeedingContext()
     }
 }
+class SubWaver : Waver {
+    func wave(reason: String) -> String {
+        return "Substitute waving"
+    }
+}
 
-
-
-
+class SubstituteContext : SimpleReferenceContext {
+    override init() {
+        super.init()
+        waver = SubWaver()
```

Comment:
As noted in the summary, having to subclass feels icky and rules out the use of struct-based contexts. It also requires class members to be mutable.